x/interchainstaking/keeper/keeper.go

```diff
@@ -201,6 +201,53 @@ func (k *Keeper) AllPortConnections(ctx sdk.Context) (pcs []types.PortConnection
 	return pcs
 }
 
+// GetZoneAndUpdateMappingIfNeeded returns zone by denom.
+func (k *Keeper) GetZoneAndUpdateMappingIfNeeded(ctx sdk.Context, denom string) *types.Zone {
```

Comment:
Can we call this `GetZoneByLocalDenom` please, as it's a little more intuitive. Otherwise, looks great!